src/k8s/pkg/k8sd/api/snap_refresh.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	apiv1 "github.com/canonical/k8s-snap-api/api/v1"
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/k8s/pkg/log"
+	"github.com/canonical/k8s/pkg/utils"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microcluster/v3/state"
+)
+
+func (e *Endpoints) postSnapRefresh(s state.State, r *http.Request) response.Response {
+	req := apiv1.SnapRefreshRequest{}
+	if err := utils.NewStrictJSONDecoder(r.Body).Decode(&req); err != nil {
+		return response.BadRequest(fmt.Errorf("failed to parse request: %w", err))
+	}
+
+	refreshOpts, err := types.RefreshOptsFromAPI(req)
+	if err != nil {
+		return response.BadRequest(fmt.Errorf("invalid refresh options: %w", err))
+	}
+	log := log.FromContext(e.Context()).WithValues("to", refreshOpts)
+
+	readyCh := make(chan error)
+	go func() {
```

Comment:
I think this part justifies a bit of commentary. 

```
// A snap refresh requires restarting services, including `k8sd`.
// If the refresh is executed during the request, it would restart the server and terminate the request, leading to an `EOF` error on the client side.
// To prevent this, the refresh is executed in a separate goroutine that waits until the response has been sent to the client.
// This approach means that the client will receive a successful response even if the subsequent refresh fails.
```

We don't inform the user somehow that the refresh actually succeeded, do we?